app/models/message.rb

```diff
@@ -5,7 +5,15 @@ class Message < ApplicationRecord
 
   attribute :content, :string, default: ''
 
-  has_many_attached :files
+  has_many :documents, dependent: :destroy
 
   enum role: %w[user assistant system].index_by(&:to_sym), _suffix: true
+
+  def attach(documents_to_attach)
+    documents_to_attach.each do |document|
+      next unless document.present?
+
+      documents.create!(documentable: UploadedFile.create!).attachment.attach(document)
+    end
+  end
```

Comment:
Suggest putting this in a transaction? So that we aren't left in a halfway state if things fail.